authui/src/errors.ts

```diff
@@ -0,0 +1,23 @@
+// Modified from https://github.com/sindresorhus/is-network-error/blob/main/index.js
```

Comment:
I see the original work has an extra checking for Safari 17+ to check if the error has an undefined stack. I guess we need to port that as well.

Reply:
In my testing that check incorrectly filter out network errors in safari (stack is not undefined), so I removed that check.

Reply:
Fair. Can you include this test as comment to explain this is one of the modification? It helps us to understand why the code is different from the original work.

Reply:
Let me do this for you.